app/react/V2/Routes/Settings/Thesauri/specs/Thesauri.spec.tsx

```diff
@@ -0,0 +1,469 @@
+          expect(rows.length).toBe(4);
+        });
+      });
+
+      it('should require the name of the group', async () => {
+        await act(async () => {
+          fireEvent.click(screen.getByText('Add group').parentNode!);
+        });
+        let newItemForm: HTMLElement;
+        await act(async () => {
+          [, newItemForm] = renderResult.container.getElementsByTagName('form');
+          fireEvent.click(screen.getByTestId('thesaurus-form-submit'));
+        });
+        await act(async () => {
+          expect(screen.getAllByText('This field is required')).toHaveLength(2);
+          fireEvent.click(within(newItemForm).getByTestId('thesaurus-form-cancel'));
+        });
+      });
+
+      it('should add items into an existing group', async () => {
+        await act(async () => {
+          fireEvent.click(screen.getByText('Add item').parentNode!);
+        });
+        await act(async () => {
+          const newItemForm = renderResult.container.getElementsByTagName('form')[1];
+          fireEvent.change(within(newItemForm).getByLabelText('Title') as HTMLInputElement, {
+            target: { value: 'new child 2' },
+          });
+          fireEvent.click(within(newItemForm).getAllByLabelText('Group')[0]);
+        });
+        await act(async () => {
+          fireEvent.change(screen.getByLabelText('Group'), {
+            target: {
+              value: (screen.getAllByRole('option')[1] as HTMLInputElement).value,
+            },
+          });
+          fireEvent.click(screen.getByTestId('thesaurus-form-submit'));
+        });
+        await act(async () => {
+          rows = await waitFor(() => screen.getAllByRole('row'));
+          expect(rows.length).toBe(4);
+        });
+      });
+
+      it('should prevent leaving without saving', async () => {
+        fireEvent.click(screen.getByText('Cancel').parentNode!);
+        await act(async () => {
+          expect(screen.getByText('Discard changes')).toBeInTheDocument();
+          fireEvent.click(within(screen.getByTestId('modal')).getByText('Cancel').parentNode!);
+        });
+      });
+
+      it('should save the new thesaurus', async () => {
+        await act(async () => {
+          fireEvent.click(screen.getByText('Save').parentNode!);
+        });
+        await act(async () => {
+          expect(saveFn).toHaveBeenCalledWith(
+            {
+              name: 'new thesaurus',
+              values: [
+                {
+                  label: 'single value 1',
+                },
+                {
+                  label: 'single value 2',
+                },
+                {
+                  label: 'Group 1',
+                  values: [
+                    {
+                      label: 'Child 1',
+                    },
+                    {
+                      label: 'new child 2',
+                    },
+                  ],
+                },
+              ],
+            },
+            undefined
```

Comment:
Is this undefined in the expect correct?

Reply:
It was expected because the headers are not passed to the calls of the API from the component, however, the methods are not called anywhere else so I removed the param in the API client.

Reply:
Ok, if we are sure those headers are never passed, we can remove them.